### PR TITLE
My Roles Call-To-Action

### LINKED ISSUE #425: My Roles: CTA for check mark and x
# Feature Request

**As a** peoplemover user
**I would** like a neater experience when creating roles
**so that** I am not confused by several line breaks (currently)

## Acceptance Criteria

**Given** I am in a space
**When** I click My Roles
**Then** I see the new My Roles Modal
**And** when I click on edit icon, two boxes appear around role name and color
**And** all of the edit & trash icons associated with other roles disappear when I am in edit mode
**And** an X and check mark icon appear on the right side of the role
**And** the X icon exits out of edit mode without saving
**And** the check icon saves the changes to the role
**And** when I leave edit mode, the edit icon and trash icon appear for the other roles

## Design
https://www.figma.com/file/S4XKeovlQqeRk64k2qzhf8gL/PeopleMover?node-id=5297%3A0

### PR DESCRIPTION
**Pull request by michaelrygiel**
## Issue
Resolves #425 

## What was done
- [x] Added edit and delete icons
- [x] Changed styling to match figma design

## How to test
1. Go to my roles modal and see updated edit and delete features for roles

Feature URL: https://featurebranchui.apps.pd01i.edc1.cf.ford.com/

### CODE REVIEW

File: ui/src/Traits/MyTraits.scss [hunk 1]
```diff
@@ -187,3 +187,36 @@
   font-size: 14px;
   margin-bottom: 6px;
 }
+
+
+
```
Comment (willot) [inline ui/src/Traits/MyTraits.scss:192]: remove crazy line and empty lines

**Review by willot — DISMISSED** [review]
it works

**Review by sabdul14 — DISMISSED** [review]

**Review by smle225 — CHANGES_REQUESTED** [review]
- Add top paddings on both icons that way they are vertically centered

**Review by smle225 — CHANGES_REQUESTED** [review]
Both icon for edit and trash can on My Tags modal are not aligned with you hit edit. Both icons for edit and trashcan are skewed to left. Let me know when you need elaboration

File: ui/src/Traits/MyTraits.tsx [hunk 2]
```diff
@@ -204,39 +204,50 @@ function MyTraits({
         }
     }
 
+    function ViewTraitRow({ trait, index }: { trait: Trait; index: number }): JSX.Element {
+        let colorToUse: string | undefined;
+        if (colorSection) {
+            const spaceRole: SpaceRole = trait as SpaceRole;
+            colorToUse = spaceRole.color ? spaceRole.color.color : '#FFFFFF';
+        }
+        const testIdTraitName = traitName.replace(' ', '');
```
[the remark this answers is not in the document]
Comment (RandomAshes) [reply ui/src/Traits/MyTraits.tsx:213]: We created this method for test id's to format them if they have a name value in them: https://github.com/FordLabs/PeopleMover/blob/stage/ui/src/tests/TestUtils.tsx#L43

The format then ends up being `yourPrefix__random_name`

Please use that to parse the data test id's :)

**Pull request merged**